### PR TITLE
[Core] Read STL file

### PR DESCRIPTION
**Pull request by qiukailu**
**📝 Description**

This PR adds a IO in order to IO STL files.

**🆕 Changelog**

- [Adding StlIO class](https://github.com/KratosMultiphysics/Kratos/pull/10697/commits/0f0bd42000a88593654efd032f7d53f61110c1aa)
- [Adding stl_io.cpp to Cmake lists](https://github.com/KratosMultiphysics/Kratos/pull/10697/commits/2954a8bd2c76befca75cc64c22438e7e8af72c8c)
- [Adding stl io python interface](https://github.com/KratosMultiphysics/Kratos/pull/10697/commits/a39a50319e937daaad1d46919738cadd437ed775)
- [Adding test for stl io](https://github.com/KratosMultiphysics/Kratos/pull/10697/commits/451886216bed30d6055ef6b211529549ca1b03ce)

### CODE REVIEW

**Comment by qiukailu** [comment]
I found this branch of Pooyan's. Although it's kind of old, it looks clean. I am planning on making this up-to-date and merging it to master.

@pooyan-dadvand @ddiezrod @pablobecker

**Comment by qiukailu** [comment]
And apparently @loumalouomega is helping me :D Thanks!

**Comment by qiukailu** [comment]
> Now it is even cleaner

I am amazed! How do you manage to make improvements so quickly? Do you have some static analysis/code formatter utility handy?

**Comment by loumalouomega** [comment]
> > Now it is even cleaner
> 
> I am amazed! How do you manage to make improvements so quickly? Do you have some static analysis/code formatter utility handy?

I am just fast :P

File: kratos/input_output/stl_io.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+void StlIO::ReadSolid(ModelPart & rThisModelPart)
+{
+    std::string word;
+
+    *mpInputStream >> word; // Reading solid or eof
+    if(mpInputStream->eof())
+        return;
+
+    KRATOS_ERROR_IF(word != "solid") << "Invalid stl file. Solid block should begin with \"solid\" keyword but \"" << word << "\" was founded" << std::endl;
```
Comment (matekelemen) [inline kratos/input_output/stl_io.cpp]: please be considerate of the mental health of english speakers
`find . -name "stl*" -type f -exec sed -i 's/founded/found/g' {} \;`

File: kratos/input_output/stl_io.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+// Project includes
+#include "includes/define.h"
+#include "input_output/stl_io.h"
+
+namespace Kratos
+{
+
+/// Constructor with  filenames.
+StlIO::StlIO(std::string const& Filename)
```
Comment (philbucher) [inline kratos/input_output/stl_io.cpp]: Please use filesystem::path instead, see the ModelPartIO

Comment (loumalouomega) [reply kratos/input_output/stl_io.cpp]: Done

**Review by loumalouomega — APPROVED** [review]
This is ready @qiukailu

**Pull request merged**